src/Common/src/TypeSystem/Common/FieldDesc.FieldLayout.cs

```diff
@@ -43,7 +43,7 @@ public virtual bool HasGCStaticBase
             get
             {
                 if (!FieldType.IsValueType)
-                    return true;
+                    return !FieldType.IsPointer;
```

Comment:
Or maybe `ContainsGCPointers`? (And rename `IsObjRef` to `IsGCPointer` to match?)